ui/src/wallet.ts

```diff
@@ -30,50 +34,72 @@ export type State =
 
 export interface Connected {
   account: Account;
+  network: ethereum.Network;
 }
 
 export interface Account {
   address: string;
-  balance: string;
+  balance: Big;
 }
 
 export interface Wallet extends svelteStore.Readable<State> {
   connect(): Promise<void>;
   disconnect(): Promise<void>;
+  provider: ethers.providers.Provider;
   signer: ethers.Signer;
   account(): Account | undefined;
 }
 
-export const provider = new ethers.providers.JsonRpcProvider(
-  "http://localhost:8545"
-);
-
-export function build(): Wallet {
-  const stateStore = svelteStore.writable<State>({
-    status: Status.NotConnected,
-  });
-
-  const qrCodeModal = {
-    open: (uri: string, _cb: unknown, _opts?: unknown) => {
-      uriStore.set(uri);
-      modal.toggle(path.walletQRCode());
-    },
-    close: async () => {
-      // N.B: this is actually called when the connection is established,
-      // not when the modal is closed per se.
-      modal.hide();
-    },
-  };
+function getProvider(
+  environment: ethereum.Environment
+): ethers.providers.Provider {
+  switch (environment) {
+    case ethereum.Environment.Local:
+      return new ethers.providers.JsonRpcProvider("http://localhost:8545");
+    case ethereum.Environment.Ropsten:
+      return new ethers.providers.InfuraProvider(
+        "ropsten",
+        "66fa0f92a54e4d8c9483ffdc6840d77b"
```

Comment:
:information_source: This is the project id that Infura provides in the backend. We have a free account so it's fine to have it here. We can always create a new project when we need to be more cautious.